### PR TITLE
Separate out refund policy for new domain registrations and renewals

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

This change implements separate wording for the domain refund policy that applies based on whether the domain is a new registration or a renewal, instead of language that applies to both situations.

#### Testing instructions

Using both the existing and new checkout flows, get to the main checkout UI for the following flows (you don't need to complete the purchase):
* Test buying a new domain with a plan; the checkout terms should include the following text:
   * You understand that [domain name refunds](https://wordpress.com/support/manage-purchases/#refund-policy) are limited to 96 hours after registration.
* Test buying a new domain as part of a plan purchase (i.e. via NUX); the checkout terms should include the following text:
   * You understand that [domain name refunds](https://wordpress.com/support/manage-purchases/#refund-policy) are limited to 96 hours after registration. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.
* Test renewing a domain; the checkout terms should include the following text:
   * Please note: to receive a [refund for a domain renewal](https://wordpress.com/support/manage-purchases/#refund-policy), you must [cancel your domain](https://wordpress.com/support/domains/#cancel-a-domain-name) within 96 hours of the renewal transaction. Canceling the domain means it will be deleted and you may not be able to recover it.

In each case, verify that the links work and link to the appropriate documentation.

_Note:_ You should be able to trigger the composite checkout by adding `?flags=composite-checkout-force` to your URL.

**Note:** I believe we don't support renewals and new registrations in the same cart at present, but the logic ought to show both messages in cases where both types of purchase are present.

updated_domain_renewal_terms

### CODE REVIEW

**Comment by sirbrillig** [comment]
Testing results:

#### Cart contains plan and domain. For both new and old checkout, I see:

> You understand that domain name refunds are limited to 96 hours after registration. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.

#### Cart includes just domain (already own a plan).

> You understand that domain name refunds are limited to 96 hours after registration.

#### Cart includes just a domain renewal.

> Please note: to receive a refund for a domain renewal, you must cancel your domain within 96 hours of the renewal transaction. Canceling the domain means it will be deleted and you may not be able to recover it.

Links appear to work correctly. 👍

File: client/my-sites/checkout/checkout/domain-refund-policy.jsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import Gridicon from 'components/gridicon';
+import {
+	hasDomainBeingUsedForPlan,
+	hasDomainRegistration,
+	hasDomainRenewal,
+	hasNewDomainRegistration,
+} from 'lib/cart-values/cart-items';
+
+class DomainRefundPolicy extends React.Component {
+	static displayName = 'RegistrationRefundPolicy';
```
Comment (klimeryk) [inline client/my-sites/checkout/checkout/domain-refund-policy.jsx:21]: I guess we can skip this line - the name of the component should be picked up from the default export. And it's confusing to have different names for this component like this anyway.

File: client/my-sites/checkout/checkout/domain-refund-policy.jsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			{
+				components: {
+					refundsSupportPage: refundsSupportPage,
+				},
+			}
+		);
+
+		if ( hasDomainBeingUsedForPlan( this.props.cart ) ) {
+			message = this.props.translate(
+				'You understand that {{refundsSupportPage}}domain name refunds{{/refundsSupportPage}} are limited to 96 hours after registration. Refunds of paid plans will deduct the standard cost of any domain name registered within a plan.',
```
Comment (klimeryk) [inline client/my-sites/checkout/checkout/domain-refund-policy.jsx:69]: Nice addition! Though, I'm not sure about the wording: `any domain name registered within a plan.`. Did you mean `with a plan`? Also, hopefully the user will understand that that applies also to the domain credit coming from the plan - but I don't think we need to say that explicitly.

File: client/my-sites/checkout/checkout/domain-refund-policy.jsx (hunk 1 above)
Comment (daledupreez) [reply client/my-sites/checkout/checkout/domain-refund-policy.jsx:21]: This is a leftover from the old code. Will remove.

File: client/my-sites/checkout/checkout/domain-refund-policy.jsx (hunk 2 above)
Comment (daledupreez) [reply client/my-sites/checkout/checkout/domain-refund-policy.jsx:69]: This wasn't me! 😁 This was already in the `DomainRegistrationRefundPolicy` code I deleted. So for now, I don't want to change that in this PR -- we can flag a revisit separately.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3683447

Hi @daledupreez, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Please note: to receive a {{refundsSupportPage}}refund for a domain renewal{{/refundsSupportPage}}, you must {{cancelDomainSupportPage}}cancel your domain{{/cancelDomainSupportPage}} within 96 hours of the renewal transaction. Canceling the domain means it will be deleted and you may not be able to recover it.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3683447).

**Pull request merged**